qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,591 @@
+        _add_edges_for_term(edge_matrix, _operator_string(term))
+    return edge_matrix
+
+
+def _add_one_edge(edge_matrix: np.ndarray, i: int, j: int) -> None:
+    """Add an edge in the adjacency matrix from lesser index to greater.
+
+    This maintains the upper triangular structure.
+
+    Args:
+      i: the first vertex index
+      j: the second vertex index. The values of `i` and `j` may be in any order,
+         but they must not be equal.
+    """  # pylint: disable=missing-raises-doc
+    if i == j:
+        raise ValueError("expecting i != j")
+    edge_matrix[min(i, j), max(i, j)] = True
+
+
+def _add_edges_for_term(edge_matrix: np.ndarray, term_str: str) -> None:
+    """Add one, two, or no edges to `edge_matrix` as dictated by the operator `term_str`.
+
+    Args:
+      edge_matrix: an adjacency matrix representing the connectivity graph.
+      term_str: A representation of the interaction satisfying the description in the method
+        `_unpack_term`. The number of explicit raising and lowering operators ('+' and '-')
+        must be equal.
+    """  # pylint: disable=missing-raises-doc
+    (n_number, n_raise, n_lower), facs = _unpack_term(term_str)
+    _type = _interaction_type(n_number, n_raise, n_lower)
+    # For EXCITATION and NUMBER_EXCITATION, create an edge between the `+` and `-`.
+    if _type in (TermType.EXCITATION, TermType.NUMBER_EXCITATION):
+        inds = [i for (i, c) in facs if c in "+-"]
+        if len(inds) != 2:
+            raise ValueError("wrong number or raising and lowering: ", len(inds))
+        _add_one_edge(edge_matrix, *inds)
+    # For `double_excitation` create an edge between the two `+`s and an edge between the two `-`s.
+    elif _type == TermType.DOUBLE_EXCITATION:
+        raise_inds = [i for (i, c) in facs if c == "+"]
+        lower_inds = [i for (i, c) in facs if c == "-"]
+        _add_one_edge(edge_matrix, *raise_inds)
+        _add_one_edge(edge_matrix, *lower_inds)
+
+
+def _bksf_edge_list_fermionic_op(ferm_op: FermionicOp) -> np.ndarray:
+    """Construct edge list required for the BKSF algorithm.
+
+    Args:
+        ferm_op: the fermionic operator in the second quantized form
+
+    Returns:
+        numpy.ndarray: edge_list, a 2xE matrix, where E is total number of edges
+                        and each pair denotes (from, to)
+    """
+    edge_matrix = _get_adjacency_matrix(ferm_op)
+    edge_list_as_2d_array = np.asarray(np.nonzero(edge_matrix))
+    return edge_list_as_2d_array
+
+
+def _edge_operator_aij(edge_list: np.ndarray, i: int, j: int) -> SparsePauliOp:
+    """Calculate the edge operator A_ij.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list: a 2xE matrix, where E is total number of edges
+                                    and each pair denotes (from, to)
+        i: specifying the edge operator A
+        j: specifying the edge operator A
+
+    Returns:
+        Pauli: qubit operator
```

Comment:
Normally the return type is not given here - in this case it seems to conflict with what the method states i.e. SparsePauliOp

Reply:
* "qubit operator" is too generic. Better might be "the edge operator A_ij". I did not say "the qubit operator as a `SparsePauliOp`, because I would then be repeating information that's already in the type annotation.
* Maybe a good solution that satisfies the letter and spirit of the docstring style guide is "Return the edge operator A_ij" as the first line of the docstring. Then omit the "Returns:" section. And also add a pylint hint to disable the complaint about the missing section.

Reply:
I revised the docstring according to my comment above.